src/shared/strings.ts

```diff
@@ -50,6 +51,10 @@ export function diffInMinutes(from: Date | string, to: Date | string): number {
   return Math.floor((new Date(to).getTime() - new Date(from).getTime()) / 1000 / 60);
 }
 
+export function compareDate(d1: string | undefined, d2: string | undefined) {
+  return moment(d1 || "").isAfter(moment(d2 || "")) ? -1 : 1;
```

Comment:
the logic should be reversed to comply with the convention of the `compare` functions